### PR TITLE
Updates to adtech amp-ad functionality #3462

### PR DESCRIPTION
**Pull request by lisamayo**
Ref: issue #3462

### CODE REVIEW

**Comment by lisamayo** [comment]
CLA has been signed as AOL.

File: ads/adtech.js [hunk 1]
```diff
@@ -14,15 +14,36 @@
...
 export function adtech(global, data) {
-  const src = data.src;
-  validateSrcPrefix('https:', src);
-  validateSrcContains('/addyn/', src);
-  writeScript(global, src);
-}
+
+ const adsrc = data.src;
+ if (typeof adsrc != 'undefined') {
+ 	validateSrcPrefix('https:', adsrc);
```
Comment (cramforce) [inline ads/adtech.js]: Indent 2

File: ads/adtech.js [hunk 2]
```diff
@@ -14,15 +14,36 @@
...
+	global.atwothat = data.atwothat;
+	global.atwplid = data.atwplid;
+	global.atwsizes = data.atwsizes;
+	global.atwwidth = data.atwwidth;
+
+  writeScript(global,'https://s.aolcdn.com/os/ads/adsWrapper3.js');
+
+   }
+
+}
```
Comment (cramforce) [inline ads/adtech.js]: Missing newline

File: ads/adtech.js [hunk 3]
```diff
@@ -14,15 +14,36 @@
...
+	global.atwmoat = data.atwmoat;
+	global.atwnetid = data.atwnetid;
+	global.atwothat = data.atwothat;
+	global.atwplid = data.atwplid;
+	global.atwsizes = data.atwsizes;
+	global.atwwidth = data.atwwidth;
+
+  writeScript(global,'https://s.aolcdn.com/os/ads/adsWrapper3.js');
+
+   }
```
Comment (cramforce) [inline ads/adtech.js]: Indent seems off.

File: ads/adtech.js [hunk 4]
```diff
@@ -14,15 +14,36 @@
...
+	global.atwhtnmat = data.atwhtnmat;
+	global.atwmn = data.atwmn;
+	global.atwmoat = data.atwmoat;
+	global.atwnetid = data.atwnetid;
+	global.atwothat = data.atwothat;
+	global.atwplid = data.atwplid;
+	global.atwsizes = data.atwsizes;
+	global.atwwidth = data.atwwidth;
+
+  writeScript(global,'https://s.aolcdn.com/os/ads/adsWrapper3.js');
```
Comment (cramforce) [inline ads/adtech.js]: Any chance you could switch to `loadScript`?

File: ads/adtech.js [hunk 5]
```diff
@@ -14,15 +14,36 @@
...
+ 	validateSrcPrefix('https:', adsrc);
+ 	validateSrcContains('/addyn/', adsrc);
+ 	writeScript(global, adsrc);
+
+ } else {
+
+  checkData(data, ['atwco', 'atwdiv', 'atwheight', 'atwhtnmat', 'atwmn', 'atwmoat', 'atwnetid', 'atwothat', 'atwplid', 'atwsizes','atwwidth']);
+
+  validateDataExists (data,['atwmn', 'atwdiv']);
+	global.atwco = data.atwco;
```
Comment (cramforce) [inline ads/adtech.js]: Indent -4

File: ads/adtech.js [hunk 6]
```diff
@@ -14,15 +14,36 @@
...
+ if (typeof adsrc != 'undefined') {
+ 	validateSrcPrefix('https:', adsrc);
+ 	validateSrcContains('/addyn/', adsrc);
+ 	writeScript(global, adsrc);
+
+ } else {
+
+  checkData(data, ['atwco', 'atwdiv', 'atwheight', 'atwhtnmat', 'atwmn', 'atwmoat', 'atwnetid', 'atwothat', 'atwplid', 'atwsizes','atwwidth']);
+
+  validateDataExists (data,['atwmn', 'atwdiv']);
```
Comment (cramforce) [inline ads/adtech.js]: No space after function. Space after `,`

File: ads/adtech.js [hunk 7]
```diff
@@ -14,15 +14,36 @@
...
+
+ const adsrc = data.src;
+ if (typeof adsrc != 'undefined') {
+ 	validateSrcPrefix('https:', adsrc);
+ 	validateSrcContains('/addyn/', adsrc);
+ 	writeScript(global, adsrc);
+
+ } else {
+
+  checkData(data, ['atwco', 'atwdiv', 'atwheight', 'atwhtnmat', 'atwmn', 'atwmoat', 'atwnetid', 'atwothat', 'atwplid', 'atwsizes','atwwidth']);
```
Comment (cramforce) [inline ads/adtech.js]: Wrap at 80 columns

**Comment by lisamayo** [comment]
Thanks for the quick feedback, updates have been made.  Need to keep our ad call synchronous so I won't be able to switch to loadScript.

File: ads/adtech.js [hunk 8]
```diff
@@ -14,15 +14,39 @@
  * limitations under the License.
  */
 
-import {writeScript, validateSrcPrefix, validateSrcContains} from '../3p/3p';
+import {writeScript, checkData, validateSrcPrefix,
+    validateSrcContains, validateDataExists,
+    }
+    from '../3p/3p';
```
Comment (cramforce) [inline ads/adtech.js]: nit: goes onto previous line.

File: ads/adtech.js [hunk 9]
```diff
@@ -14,15 +14,39 @@
  * limitations under the License.
  */
 
-import {writeScript, validateSrcPrefix, validateSrcContains} from '../3p/3p';
+import {writeScript, checkData, validateSrcPrefix,
```
Comment (cramforce) [inline ads/adtech.js]: Please format like an object literal (one line per entry, 2 space indent).

File: ads/adtech.js [hunk 10]
```diff
@@ -14,15 +14,42 @@
...
+    global.atwheight = data.atwheight;
+    global.atwhtnmat = data.atwhtnmat;
+    global.atwmn = data.atwmn;
+    global.atwmoat = data.atwmoat;
+    global.atwnetid = data.atwnetid;
+    global.atwothat = data.atwothat;
+    global.atwplid = data.atwplid;
+    global.atwsizes = data.atwsizes;
+    global.atwwidth = data.atwwidth;
+    writeScript(global,'https://s.aolcdn.com/os/ads/adsWrapper3.js');
```
Comment (cramforce) [inline ads/adtech.js]: Nit: space after ,

File: ads/adtech.js [hunk 11]
```diff
@@ -14,15 +14,42 @@
...
-  const src = data.src;
-  validateSrcPrefix('https:', src);
-  validateSrcContains('/addyn/', src);
-  writeScript(global, src);
+  const adsrc = data.src;
+  if (typeof adsrc != 'undefined') {
+    validateSrcPrefix('https:', adsrc);
+    validateSrcContains('/addyn/', adsrc);
+    writeScript(global, adsrc);
+
```
Comment (cramforce) [inline ads/adtech.js]: nit: remove line

File: ads/adtech.js [hunk 12]
```diff
@@ -14,15 +14,42 @@
...
+  } else {
+
```
Comment (cramforce) [inline ads/adtech.js]: Nit: remove line

File: ads/adtech.js [hunk 13]
```diff
@@ -14,15 +14,42 @@
...
+    checkData(data, [
+      'atwco', 'atwdiv', 'atwheight', 'atwhtnmat',
+      'atwmn', 'atwmoat', 'atwnetid', 'atwothat', 'atwplid',
+      'atwsizes','atwwidth',
+    ]);
+    validateDataExists(data,['atwmn', 'atwdiv']);
```
Comment (cramforce) [inline ads/adtech.js]: Nit: space after ,

**Comment by cramforce** [comment]
Almost ready to go.

**Comment by cramforce** [comment]
LGTM

**Comment by lisamayo** [comment]
Thank you!

**Pull request merged**